packages/babel-plugin-transform-classes/src/transformClass.js

```diff
@@ -146,20 +146,17 @@ export default function transformClass(
    * Creates a class constructor or bail out if there is none
    */
   function maybeCreateConstructor() {
-    let hasConstructor = false;
     const paths = classState.path.get("body.body");
     for (const path of paths) {
-      hasConstructor = path.equals("kind", "constructor");
-      if (hasConstructor) break;
+      if (path.equals("kind", "constructor")) return;
     }
-    if (hasConstructor) return;
 
     let params, body;
 
     if (classState.isDerived) {
       const constructor = template.expression.ast`
-        (function () {
-          super(...arguments);
+        (function (...args) {
```

Comment:
An issue description would be helpful. What's the motivation for this change?

Reply:
Whoops, fixes #5147.

Reply:
I think that in loose mode we should keep using `arguments`, since it produces a smaller (and probably faster) output.